client.cpp

```diff
@@ -878,6 +878,130 @@ bool verify_client::finished(void)
 
 ///////////////////////////////////////////////////////////////////////////
 
+crc_verify_client::verify_request::verify_request(request_type type,
+                                              unsigned int size,
+                                              struct timeval* sent_time,
+                                              unsigned int keys,
+                                              const char *key,
+                                              unsigned int key_len) :
+        client::request(type, size, sent_time, keys),
+        m_key(NULL), m_key_len(0)
+{
+    m_key_len = key_len;
+    m_key = (char *)malloc(key_len);
+    memcpy(m_key, key, m_key_len);
+}
+
+crc_verify_client::verify_request::~verify_request(void)
+{
+    if (m_key != NULL) {
+        free((void *) m_key);
+        m_key = NULL;
+    }
+}
+
+crc_verify_client::crc_verify_client(struct event_base *event_base,
+                             benchmark_config *config,
+                             abstract_protocol *protocol,
+                             object_generator *obj_gen) : client(event_base, config, protocol, obj_gen),
+                                                          m_finished(false), m_verified_keys(0), m_errors(0)
+{
+    m_protocol->set_keep_value(true);
+    m_obj_gen->set_key_range(config->key_minimum, config->key_maximum);
+}
+
+unsigned long long int crc_verify_client::get_verified_keys(void)
+{
+    return m_verified_keys;
+}
+
+unsigned long long int crc_verify_client::get_errors(void)
+{
+    return m_errors;
+}
+
+void crc_verify_client::create_request(struct timeval timestamp)
+{
+    // Prepare a GET request that will be compared against a previous
+    // SET request.
+    unsigned int cmd_size;
+    int iter = obj_iter_type(m_config, 2);
```

Comment:
what does `2` mean here?

Reply:
It's an ugly magic used in different places in the code. The obj_iter_type receives an index arg and uses it to access the config key_pattern string. This function is used to determine the key pattern to use. It can be re-factored but it would require touching other places in the code that I don't want to touch right now.

Reply:
dont they have defines for that ?

Reply:
No, they have obj_iter_type(m_config, 2) and obj_iter_type(m_config, 0) calles in different places, no defines are used for that